src/main/java/com/google/ehub/data/EntertainmentItemDatastore.java

```diff
@@ -12,54 +13,97 @@
+   *
+   * @return single instance of the class
+   */
+  public static EntertainmentItemDatastore getInstance() {
+    if (instance == null) {
+      instance = new EntertainmentItemDatastore();
+    }
 
-  private static DatastoreService datastoreService = DatastoreServiceFactory.getDatastoreService();
+    return instance;
+  }
 
   /**
    * Adds an EntertainmentItem to Datastore.
    *
-   * @param title title of the entertainment item
-   * @param description description of the entertainment item
-   * @param imageURL URL to the image of the entertainment item stored in Blobstore
+   * @param entertainmentItem entertainment item to add into Datastore
    */
-  public static void addEntertainmentItemToDatastore(
-      String title, String description, String imageURL) {
+  public void addItemToDatastore(EntertainmentItem entertainmentItem) {
     Entity entertainmentItemEntity = new Entity(ENTERTAINMENT_ITEM_KIND);
 
-    entertainmentItemEntity.setProperty(TITLE_PROPERTY_KEY, title);
-    entertainmentItemEntity.setProperty(LOWERCASE_TITLE_PROPERTY_KEY, title.toLowerCase());
-    entertainmentItemEntity.setProperty(DESCRIPTION_PROPERTY_KEY, description);
-    entertainmentItemEntity.setProperty(IMAGE_URL_PROPERTY_KEY, imageURL);
+    entertainmentItemEntity.setProperty(TITLE_PROPERTY_KEY, entertainmentItem.getTitle());
+    entertainmentItemEntity.setProperty(
+        LOWERCASE_TITLE_PROPERTY_KEY, entertainmentItem.getTitle().toLowerCase());
+    entertainmentItemEntity.setProperty(
+        DESCRIPTION_PROPERTY_KEY, entertainmentItem.getDescription());
+    entertainmentItemEntity.setProperty(IMAGE_URL_PROPERTY_KEY, entertainmentItem.getImageURL());
 
     datastoreService.put(entertainmentItemEntity);
   }
 
+  /**
+   * Finds a single entertainment item based on unique id.
+   *
+   * @param uniqueID id used to identify the EntertainmentItem Entity in the Datastore
+   * @return the EntertainmentItem found in Datastore wrapped in an {@link Optional}, the
+   * optional object will be empty if the EntertainmentItem Entity was not found
+   */
+  public Optional<EntertainmentItem> querySingleItem(long uniqueID) {
+    Query query =
+        new Query(ENTERTAINMENT_ITEM_KIND)
+            .setFilter(new FilterPredicate(Entity.KEY_RESERVED_PROPERTY, FilterOperator.EQUAL,
+                KeyFactory.createKey(ENTERTAINMENT_ITEM_KIND, uniqueID)));
+    PreparedQuery queryResults = datastoreService.prepare(query);
+
+    Entity entertainmentItemEntity = queryResults.asSingleEntity();
+
+    if (entertainmentItemEntity == null) {
+      return Optional.empty();
+    }
+
+    return Optional.of(getItemFromEntity(entertainmentItemEntity));
+  }
+
   /**
    * Queries all entertainment items found in Datastore.
    *
    * @return list with all entertainment items found in Datastore
    */
-  public static List<EntertainmentItem> queryAllEntertainmentItems() {
+  public List<EntertainmentItem> queryAllEntertainmentItems() {
```

Comment:
nit: remove "Entertainment" from function name since it's not being used above

Reply:
I removed references to the term "Entertainment" in the methods of EntertainmentItemDatastore class in last commit.